File: IPWorks Samples/Atom Feed Reader/python/atom_feed_reader.py
# 
# IPWorks 2022 Python Edition - Sample Project
# 
# This sample project demonstrates the usage of IPWorks in a 
# simple, straightforward way. It is not intended to be a complete 
# application. Error handling and other checks are simplified for clarity.
# 
# www.nsoftware.com/ipworks
# 
# This code is subject to the terms and conditions specified in the 
# corresponding product license agreement which outlines the authorized 
# usage and restrictions.
# 

import sys
import string
from ipworks import *

input = sys.hexversion<0x03000000 and raw_input or input


def fireError(e):
  print("ERROR: %s" %e.description)
  
def fireSSLServerAuthentication(e):
  e.accept = True

try:
  atom = Atom()

  atom.on_error = fireError
  atom.on_ssl_server_authentication = fireSSLServerAuthentication

  print("This demo uses the most recent Google News atom feed. Sometimes Google changes the url so if you get an error try updating the url below")
  feed = "https://news.google.com/atom?hl=en-US&gl=US&ceid=US:en"
  atom.get_feed(feed)

  while True:
    print("Atom Channel :%s\n%s\n"%(atom.get_channel_title(),atom.get_channel_subtitle()))
    for i in range(0,atom.get_entry_count()):
      print("%i, %s\n"%(i+1,atom.get_entry_title(i)))
    print("Q, Quit\n")
    response = (input("news item number: ")).lower()
    if response.startswith('q'):
      break
    else:
      selectedindex = int(response)-1
      print("\n\n[%d] %s\n\n" %(int(response),atom.get_entry_content(selectedindex)))
      print("Find full article at %s\n\n" %atom.get_entry_link_href(selectedindex))

except IPWorksError as e:
  print("ERROR: %s" %e.message)

except KeyboardInterrupt:
  print("Shutdown requested...exiting")

File: IPWorks Samples/Atom Feed Reader/php/atom_feed_reader.php
<?php $sendBuffer = TRUE; ob_start(); ?>
<html>
<head>
<title>IPWorks 2022 Demos - Atom Feed Reader</title>
<link rel="stylesheet" type="text/css" href="stylesheet.css">
<meta name="description" content="IPWorks 2022 Demos - Atom Feed Reader">
</head>

<body>

<div id="content">
<h1>IPWorks - Demo Pages</h1>
<h2>Atom Feed Reader</h2>
<p>Shows how to use the Atom component to read Atom feeds.</p>
<a href="default.php">[Other Demos]</a>
<hr/>

<?php
require_once('../include/ipworks_atom.php');
require_once('../include/ipworks_const.php');

?>

<?php
	class MyAtom extends IPWorks_Atom
	{
		function FireSSLServerAuthentication($param) {
			$param['accept'] = true;
			return $param;
		}
	}
   
	$atom1 = new MyAtom();
?>

<?php
if ($_SERVER['REQUEST_METHOD'] == "POST") {
  $atom1->setFollowRedirects(1);
  $atom1->doGetFeed($_POST["url"]);
  echo $atom1->getChannelTitle() . " - " . $atom1->getChannelSubtitle();
?>

  <table>

<?php
  for($i = 0; $i < $atom1->getEntryCount();$i++){
        echo "<tr><td><a href=\"" . $atom1->getEntryLinkHref($i) . "\">" . $atom1->getEntryTitle($i) . "</a></td>";
        echo "<td>" . $atom1->getEntryPublished($i) . "</td></tr>";
  }
?>

  </table>

<?php
}
?>

<form method=POST>
<center>
<table width="90%">

 <tr><td>Feed: <td><input type=text name=url value="https://news.google.com/news?ned=us&topic=h&output=atom"  size=60>

 <tr><td><td><br><input type=submit value="Get Feed!">

</table>
</center>
</form>



<br/>
<br/>
<br/>
<hr/>
NOTE: These pages are simple demos, and by no means complete applications.  They
are intended to illustrate the usage of the IPWorks objects in a simple,
straightforward way.  What we are hoping to demonstrate is how simple it is to
program with our components.  If you want to know more about them, or if you have
questions, please visit <a href="http://www.nsoftware.com/?demopg-IPPHA" target="_blank">www.nsoftware.com</a> or
contact our technical <a href="http://www.nsoftware.com/support/">support</a>.
<br/>
<br/>
Copyright (c) 2023 /n software inc.
<br/>
<br/>
</div>

<div id="footer">
<center>
IPWorks 2022 - Copyright (c) 2023 /n software inc. - For more information, please visit our website at <a href="http://www.nsoftware.com/?demopg-IPPHA" target="_blank">www.nsoftware.com</a>.
</center>
</div>

</body>
</html>

<?php if ($sendBuffer) ob_end_flush(); else ob_end_clean(); ?>
